tlsn/tlsn-notary-client/src/client.rs

```diff
@@ -36,21 +47,13 @@ pub struct NotaryClient {
     /// Maximum number of bytes that can be received.
     #[builder(default = "DEFAULT_MAX_RECV_LIMIT")]
     max_recv_data: usize,
-    /// Root certificate store used for establishing TLS connection with notary
-    #[builder(default = "Some(notary_default_root_store()?)")]
-    notary_root_cert_store: Option<RootCertStore>,
-    /// DNS name of notary server used for establishing TLS connection with notary
-    #[builder(setter(into), default = "Some(\"tlsnotaryserver.io\".to_string())")]
+    /// Root certificate store used for establishing TLS connection with notary server (should be None if TLS is not used)
+    root_cert_store: Option<RootCertStore>,
+    /// Notary server DNS name (should be None if TLS is not used)
     notary_dns: Option<String>,
```

Comment:
Just tried this and got an issue, for anyone using self-signed cert (and self-named DNS), this would fail when this client tries to form connection to the `host`, as the self-named DNS will cause address lookup failure. Although its likely only our tests and examples are going to use self-signed cert, i think it's still very useful to support this from test coverage perspective of the notary server and client.

The only way I can think of so far to support this is to have an optional `ip` field that get used if it is set (and should default to None). What do you think?

Reply:
Decision after discussion: to not have notary client crate to support this use case since it's only going to be used by integration test and examples — we can still integration-test TLS setup via other mechanisms